### PR TITLE
Feature/add entry review notification

### PR DESCRIPTION
**Pull request by frozenhelium**
Depends on: https://github.com/the-deep/server/pull/704

### CODE REVIEW

File: src/components/general/Navbar/Notifica/Notifications/index.js [hunk 1]
```diff
@@ -140,6 +144,7 @@ const requestOptions = {
             props: { setNotifications },
             response: { results },
         }) => {
+            console.info(results);
```
Comment (AdityaKhatri) [inline src/components/general/Navbar/Notifica/Notifications/index.js]: remove this

Comment (frozenhelium) [reply src/components/general/Navbar/Notifica/Notifications/index.js]: oh no!

File: src/components/general/Navbar/Notifica/Notifications/items/EntryReview/index.tsx [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+                break;
+            case REVIEW_TYPE_APPROVE:
+                message = _ts('notifications.entryReviewComment', 'approve', {
+                    assigner,
+                    entry: commentText ? entryLinkItemWithReview : entryLinkItem,
+                    projectName: projectLinkItem,
+                });
+                break;
+            case REVIEW_TYPE_UNAPPROVE:
+                message = 'bla bla';
```
Comment (AdityaKhatri) [inline src/components/general/Navbar/Notifica/Notifications/items/EntryReview/index.tsx]: Please remove this

File: src/components/general/Navbar/Notifica/Notifications/items/EntryReview/index.tsx [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    } else if (notificationType === 'entry_review_comment_modify') {
+        message = _ts('notifications.entryReviewComment', 'commentUpdate', {
+            comment: commentLinkItem,
+        });
+    }
+
+    return (
+        <Notification
+            notificationId={id}
+            message={message}
```
Comment (AdityaKhatri) [inline src/components/general/Navbar/Notifica/Notifications/items/EntryReview/index.tsx:182]: There is a typescript error for the message.

File: src/components/general/Navbar/Notifica/Notifications/index.js (hunk 1 above)
Comment (frozenhelium) [reply src/components/general/Navbar/Notifica/Notifications/index.js]: Done!

File: src/components/general/Navbar/Notifica/Notifications/items/EntryReview/index.tsx (hunk 2 above)
Comment (frozenhelium) [reply src/components/general/Navbar/Notifica/Notifications/items/EntryReview/index.tsx]: Done!

File: src/components/general/Navbar/Notifica/Notifications/items/EntryReview/index.tsx (hunk 3 above)
Comment (frozenhelium) [reply src/components/general/Navbar/Notifica/Notifications/items/EntryReview/index.tsx:182]: Done!

**Review by subinasr — APPROVED** [review]

**Review by samshara — APPROVED** [review]

File: src/components/general/Navbar/Notifica/Notifications/items/EntryReview/index.tsx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+                id: number;
+                name: string;
+                email: string;
+            };
+            projectDetails: {
+                id: number;
+                title: string;
+            };
+        };
+    };
```
Comment (tnagorra) [inline src/components/general/Navbar/Notifica/Notifications/items/EntryReview/index.tsx]: Please separate this into a separate type

File: src/components/general/Navbar/Notifica/Notifications/items/EntryReview/index.tsx [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+            link={entriesPageLink}
+            title={_ts('notifications.entryReviewComment', 'commentLabel')}
+            closeModal={closeModal}
+        />,
+        <LinkItem
+            link={projectPageLink}
+            title={projectName}
+            closeModal={closeModal}
+        />,
+    ]), [entriesPageLink, entriesPageLinkWithReview, projectPageLink, projectName, closeModal]);
```
Comment (tnagorra) [inline src/components/general/Navbar/Notifica/Notifications/items/EntryReview/index.tsx]: Please don't memoize these

File: src/components/general/Navbar/Notifica/Notifications/items/EntryReview/index.tsx (hunk 4 above)
Comment (frozenhelium) [reply src/components/general/Navbar/Notifica/Notifications/items/EntryReview/index.tsx]: Done!

File: src/components/general/Navbar/Notifica/Notifications/items/EntryReview/index.tsx (hunk 5 above)
Comment (frozenhelium) [reply src/components/general/Navbar/Notifica/Notifications/items/EntryReview/index.tsx]: Done!

**Review by tnagorra — APPROVED** [review]

**Review by subinasr — APPROVED** [review]

File: src/typings/notifications.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+export type NotificationType = 'project_join_request'
+    | 'project_join_response'
+    | 'project_join_request_abort'
+    | 'entry_review_comment_add'
+    | 'entry_review_comment_modify'
+
+export interface NotificationFields {
+    id: number;
+    project: number;
+    receiver: 5;
```
Comment (AdityaKhatri) [inline src/typings/notifications.ts]: Sir what?
Is receiver only allowed to be 5?

Comment (frozenhelium) [reply src/typings/notifications.ts]: naah, should be a `number` or `UserFields['id']`

**Pull request merged**